libsolidity/codegen/CompilerContext.cpp

```diff
@@ -167,6 +167,18 @@ CompilerContext& CompilerContext::appendJump(eth::AssemblyItem::JumpType _jumpTy
 	return *this << item;
 }
 
+CompilerContext& CompilerContext::appendInvalid()
+{
+	return *this << Instruction::INVALID;
+}
+
+CompilerContext& CompilerContext::appendConditionalInvalid()
+{
+	*this << Instruction::ISZERO;
+	eth::AssemblyItem afterTag = appendConditionalJump();
+	return *this << Instruction::INVALID << afterTag;
```

Comment:
The code generator is not too smart about the stack height with regards to control flow, so you have to manually adjust the stack height at the end of this function (it assumes that control flow continues after `invalid`).